test/unit/content-negotiation.spec.ts

```diff
@@ -1,126 +1,233 @@
-        const result = isMediaTypeSupported('text/plain/html', ['text/plain/css']);
+        const result = findMatchingType('text/plain/html', ['text/plain/css']);
 
-        expect(result).toBe(true);
+        expect(result).toBeTruthy();
+    });
+});
+
+// more specific tests
+describe('#findMatchingType', () => {
+    describe('exact matches', () => {
+        it('matches none', () => {
+            expect(findMatchingType('xxx', ['aaa', 'bbb'])).toBe(undefined);
+        });
+
+        it('matches one', () => {
+            expect(findMatchingType('aaa', ['aaa', 'bbb'])).toBe('aaa');
+            expect(findMatchingType('bbb', ['aaa', 'bbb'])).toBe('bbb');
+        });
+
+        it('matches one of many', () => {
+            expect(findMatchingType('aaa,xxx', ['aaa', 'bbb'])).toBe('aaa');
+            expect(findMatchingType('xxx,aaa', ['aaa', 'bbb'])).toBe('aaa');
+            expect(findMatchingType('bbb,xxx', ['aaa', 'bbb'])).toBe('bbb');
+            expect(findMatchingType('xxx,bbb', ['aaa', 'bbb'])).toBe('bbb');
+        });
+
+        it('matches one of many, ignoring whitespaces', () => {
+            expect(findMatchingType('aaa, xxx', ['aaa', 'bbb'])).toBe('aaa');
+            expect(findMatchingType('xxx, aaa', ['aaa', 'bbb'])).toBe('aaa');
+            expect(findMatchingType('bbb, xxx', ['aaa', 'bbb'])).toBe('bbb');
+            expect(findMatchingType('xxx, bbb', ['aaa', 'bbb'])).toBe('bbb');
+        });
+    });
+
+    describe('with parameters', () => {
+        it('prefers exact matches, but falls back accordingly', () => {
+            expect(findMatchingType('aaa;v2', ['aaa', 'aaa;v2'])).toBe('aaa;v2');
+            expect(findMatchingType('aaa', ['aaa', 'aaa;v2'])).toBe('aaa');
+            expect(findMatchingType('aaa;v2', ['aaa'])).toBe('aaa');
+            expect(findMatchingType('aaa;v3', ['aaa;v1', 'aaa;v2'])).toBe('aaa;v1');
+        });
+
+        it('prefers exact matches, but falls back accordingly, ignoring whitespaces', () => {
+            expect(findMatchingType('aaa; v2', ['aaa', 'aaa;v2'])).toBe('aaa;v2');
+            expect(findMatchingType('aaa;v2', ['aaa', 'aaa; v2'])).toBe('aaa; v2');
+            expect(findMatchingType('aaa', ['aaa', 'aaa; v2'])).toBe('aaa');
+            expect(findMatchingType('aaa; v2', ['aaa'])).toBe('aaa');
+        });
+    });
+
+    describe('extensions', () => {
+        it('matches exactly', () => {
+            expect(findMatchingType('application/vnd.foo+json', ['application/json', 'application/vnd.foo+json'])).toBe(
+                'application/vnd.foo+json'
+            );
+            expect(findMatchingType('application/json', ['application/json', 'application/vnd.foo+json'])).toBe(
+                'application/json'
+            );
+        });
+
+        it('fallsback to base content-type', () => {
+            expect(findMatchingType('application/vnd.foo+json', ['application/json'])).toBe('application/json');
```

Comment:
I think this answers the above.

Reply:
Not quite. This means that it strips the vendor extension.

Reply:
Ah yes, got it.